tools/export_torchscript.py

```diff
@@ -76,5 +77,189 @@ def main():
+    img, _ = preproc(img, None, test_size)
+    img = torch.from_numpy(img).unsqueeze(0)
+    img = img.float()
+    return img.permute(0, 2, 3, 1)
+
+
+@logger.catch
+def args_main(exp_file, name, opts, decode_in_inference, batch_size, output_name, ckpt=None):
+    exp = get_exp(exp_file, name)
+    exp.merge(opts)
+
+    model = exp.get_model()
+    if ckpt is None:
+        file_name = os.path.join(exp.output_dir, name)
+        ckpt_file = os.path.join(file_name, "best_ckpt.pth")
+    else:
+        ckpt_file = ckpt
+
+    # load the model state dict
+    ckpt = torch.load(ckpt_file, map_location="cpu")
+
+    model.eval()
+    if "model" in ckpt:
+        ckpt = ckpt["model"]
+    model.load_state_dict(ckpt)
+    model.head.decode_in_inference = decode_in_inference
+    model_deploy = DeployModule(1, 0.3, 0.45, True, (416, 416))
+    model_deploy.yolox = model
+    model.eval()
+    logger.info("loading checkpoint done.")
+    dummy_input = torch.randn(batch_size, 3, exp.test_size[0], exp.test_size[1], dtype=torch.float32)
+    image_path = r'C:\Users\arusak\new zones 2\clean view\image.jpg'
+    image = get_tensor_image(image_path)
+    test_zone = torch.tensor([
+        [
+            446,
+            149
+        ],
+        [
+            501,
+            160
+        ],
+        [
+            501,
+            265
+        ],
+        [
+            482,
+            444
+        ],
+        [
+            429,
+            431
+        ],
+        [
+            399,
+            544
+        ],
+        [
+            453,
+            556
+        ],
+        [
+            439,
+            604
+        ],
+        [
+            381,
+            594
+        ],
+        [
+            377,
+            678
+        ],
+        [
+            310,
+            667
+        ],
+        [
+            298,
+            570
+        ],
+        [
+            298,
+            438
+        ],
+        [
+            205,
+            418
+        ],
+        [
+            213,
+            247
+        ],
+        [
+            362,
+            246
+        ],
+        [
+            364,
+            157
+        ]
+    ], dtype=torch.float32)
+    image = load_image(r'C:\Users\arusak\new zones 2\clean view\image.jpg')
+    image = letterbox_image(image, (416, 416))
+    image = np.expand_dims(image, axis=0)
+    image = torch.tensor(image.tolist(), dtype=torch.float32)
+    mod = torch.jit.trace(model_deploy, {"inputs": {"input": image, "zone": test_zone}})
+    with torch.no_grad():
+        outputs = mod({"inputs": {"input": image, "zone": test_zone}})
+    for output in outputs[0]:
+        print(output)
+    mod.save(output_name)
+    logger.info("generated torchscript model named {}".format(output_name))
+
+
 if __name__ == "__main__":
-    main()
+    args_main(
+        r'\\fs-ai\NCBiR_N\ComputerVision\object_detection\models\SoS2.6_NDS_extended2\yolox\SoS_2.6_NDS_extended2_s_416\yolox_s.py',
```

Comment:
please add and use paths as in CV repository:
DISK_M = "M:/" if os.name == 'nt' else '/mnt/m'
DISK_N = "N:/" if os.name == 'nt' else '/mnt/n'
DISK_O = "O:/" if os.name == 'nt' else '/mnt/o'
CV = os.path.join(DISK_N, "ComputerVision")